allensdk/brain_observatory/behavior/behavior_data_session.py

```diff
@@ -14,6 +14,17 @@
 class BehaviorDataSession(object):
     def __init__(self, api: Optional[BehaviorDataApi] = None):
         self.api = api
+        # Initialize attributes to be lazily evaluated
```

Comment:
I wonder if these could just be instantiated as a LazyProperty https://github.com/AllenInstitute/AllenSDK/blob/master/allensdk/core/lazy_property/lazy_property.py. The implementation of LazyProperty is not documented, perhaps @NileGraddis could comment whether this would do it.
Also the commit message states: "update changes the properties to attributes", but in fact, we still have properties, but now with the setter methods and lazy evaluation. So the commit message is a bit confusing to me.

Reply:
You're right, it's a confusing commit message. I will update the verbiage to properly describe them as settable, lazily evaluated properties. 

Looking at `LazyProperty`, what I don't like is that I can't write a docstring or specify the return type, which I think is really useful when looking at the online documentation. I think if we updated `LazyProperty` to be a decorator that it would fit the bill.

Reply:
after you added  the word "essentially" the verbiage is sufficiently clear, thanks for elaborating. 
Also, I am just wondering, if memoizing get_licks() (and some other functions) was sufficient to be able to change data in place, why not just memoize? And a related question why in the BehaviourOphysApi get_licks are memoized and in the BehaviourDataLimsApi they are not?

Reply:
That's a good question. The `@memoize` decorator got around the limitations of `@property` because it accessed the cache first. I wanted to make the intended behavior clear in the Session object rather than rely on caching at the API level to basically override `@property` setter limitations. 

I also didn't want the API layer to dictate whether the Session attributes/properties are settable or not. The reason those calls are memoized is because we tested them and they take a while, say because they have to query a database and retrieve a file. But if we added another API that didn't need to be memoized, say, an NWB API that pulls quickly from a local file, then suddenly that Session object's properties would no longer be settable. We want to ensure that the behavior of a Session object is the same regardless of its fetch API.

Reply:
Makes perfect sense, thanks for clarifying.